### ISSUE TITLE
Add mutator data to route entities

### ISSUE THREAD

**Issue by JeffBobbo** [red-eclipse/base#310]
Maps have different routes for endurance and non-endurance. Some maps also have routes built specially which are different for other mutators (like vampire).

**Comment by bonifarz** [comment]
Like vampire? You refer to escape ultima? Well, we only have proper routes on official race maps (as far as i know), and even those where created before endurance was available. Though, for just giving a clue where to go those are fine, no matter which mutators are played. You can have separate routes for all kind of things via route id, but then you need some way to inform players about the meaning of the different indices.

**Comment by JeffBobbo** [comment]
This is kinda to go with #309. I know they're not in official maps but that's still used in unofficial maps (e.g., tutorial). It'd also go, as I originally mentioned, for endurance. I figured down the road we could end up with a with a simple gui that lists the available routes with descriptions like "easy" or "fast".

**Comment by bonifarz** [comment]
True, and there I mention the help menu, where the current radio labels are "off", "basic" and "advanced". 

But hey, as we discuss this here and there, I get the impression that some new world variables could be useful: routename0, routename1 ... similar to the different obits we can assign to different waters and lavas. Does not matter how many, maybe 5? With this you could also parse the number of routes... loop until the next routename is an empty string.
For instance, in the new editing menus, I also added some menu for routes. As you cannot really get the intended number of routes in use, I added some buttons to select and count the entities.

tldr: route names as world vars would be nice. Either as another solution/description for this issue, or as a new feature request. What do you think?

**Comment by JeffBobbo** [comment]
@bonifarz I was thinking in the long run it might be a good idea to change how routes work to be like waypoints instead of entities, however manually editing them today has been really useful.

So, I guess we can just have routename0...routenameN, or just routename which is an array of names.

**Comment by qreeves** [comment]
More like how bases work, `route_N = [name]`

**Comment by JeffBobbo** [comment]
This seems to have run away a bit and get rather off topic.

Maintaining all these routes may become chaotic, endurance could be considered an advanced version of the normal routes for the large part I think.

Perhaps if we were going to go ahead with this, we'd need a way to hide route entities from view (or at least selection like entediting) so we don't have to worry ourselves with the other routes or potentially break them by accident.

**Comment by bonifarz** [comment]
Haha, yeah, there are now three issues for those routes, and your last point open room for another one. That being said, I don't think a mutator attribute is needed on the entity if we have support for different names as world aliases.
